src/CommandRegistry.cpp

```diff
@@ -1,31 +1,51 @@
 #include "CommandRegistry.hpp"
 
-CommandRegistry::CommandRegistry() {}
+#include "commands/CapCommand.hpp"
+#include "commands/JoinCommand.hpp"
+#include "commands/LusersCommand.hpp"
+#include "commands/MotdCommand.hpp"
+#include "commands/NickCommand.hpp"
+#include "commands/PassCommand.hpp"
+#include "commands/PingCommand.hpp"
+#include "commands/QuitCommand.hpp"
+#include "commands/UserCommand.hpp"
+
+CommandRegistry::CommandRegistry() {
+	registerCommand("CAP", new CapCommand());
+	registerCommand("NICK", new NickCommand());
+	registerCommand("PING", new PingCommand());
+	registerCommand("USER", new UserCommand());
+	registerCommand("PASS", new PassCommand());
+	registerCommand("JOIN", new JoinCommand());
+	registerCommand("QUIT", new QuitCommand());
+	registerCommand("LUSERS", new LusersCommand());
+	registerCommand("MOTD", new MotdCommand());
+}
 
 CommandRegistry::~CommandRegistry() {
 	for (std::map<std::string, Command*>::iterator it = commands.begin();
 		 it != commands.end(); it++)
 		delete it->second;
 }
 
-static std::string strtolower(const std::string& str) {
+static std::string strtoupper(const std::string& str) {
 	std::string lstr = str;
 	for (size_t i = 0; i < lstr.size(); i++)
-		lstr[i] = tolower(lstr[i]);
+		lstr[i] = toupper(lstr[i]);
 	return lstr;
 }
 
 void CommandRegistry::registerCommand(const std::string& name,
 									  Command* command) {
-	std::string lowerName = strtolower(name);
-	if (commands.find(lowerName) != commands.end())
+	std::string upperName = strtoupper(name);
+	if (commands.find(upperName) != commands.end())
 		throw DuplicateException();
-	commands.insert(std::pair<std::string, Command*>(lowerName, command));
+	commands.insert(std::pair<std::string, Command*>(upperName, command));
 }
 
 Command* CommandRegistry::getCommand(const std::string& name) const {
 	std::map<std::string, Command*>::const_iterator cmd =
-		commands.find(strtolower(name));
+		commands.find(strtoupper(name));
```

Comment:
Is it neccessary to use strtoupper if the command names are already stored in uppercase

Reply:
There was decided to accept commands either in upper or in lower case so yes